scripts/system_performance/models/tests/test_query_argument_constructor.py

```diff
@@ -0,0 +1,221 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""Unit tests  for scripts/system_performance/models/query_argument_constructor.py"""
+import unittest
+
+from datetime import date
+from mock import patch, Mock, MagicMock
+
+from scripts.system_performance.models import query_argument_constructor
+# pylint:disable=line-too-long
+from scripts.system_performance.data_persistence.system_performance_queries import DatabaseMonitorChecks  # pylint: disable=line-too-long
+from scripts.system_performance.test_setup.test_setup_default_mock_variables import SetupVariables
+
+
+class MockConnection(Mock):
+    """Mock object class"""
+    pass # pylint: disable=W0107
+
+
+class TestQueryArgumentsConstructor(unittest.TestCase):
+    """Unit tests for query argument constructor class methods"""
+
+    def setUp(self):
+        """Initial setup of default query argument dictionary """
+        self.db_monitor_checks = DatabaseMonitorChecks()
+
+        self.arguments_dict = SetupVariables().arguments_dict
+
+    @patch('utils.clients.database_client.DatabaseClient.connect', return_value=MockConnection()) # pylint: disable=no-self-use
+    # pylint: disable=no-value-for-parameter, no-self-use
+    def test_patch_applicator(self, _):
+        """Applies patches to method called inside"""
+        db_monitor_checks = DatabaseMonitorChecks()
+        db_monitor_checks.query_log_and_execute = MagicMock(name='query_log_and_execute')
+        return db_monitor_checks
+
+    def test_get_day_of_week_invalid(self):
+        """Testing invalid day of week"""
+        invalid = 10
+        actual = query_argument_constructor.get_day_of_week()
+        self.assertNotEqual(invalid, actual)
+
+    def test_get_day_of_week_valid(self):
+        """Testing valid day of week"""
+        expected = date.today()
+        actual = query_argument_constructor.get_day_of_week()
+        self.assertEqual(expected, actual)
+
+    def test_get_instruments(self):
+        """Assert that a list of instruments is returned """
+        actual = query_argument_constructor.get_instruments()
+        # expected = ['GEM', 'WISH', 'MUSR']
```

Comment:
Travis duplication warning: Instead of testing the values returned, this should just test that the expected function in `database monitor checks` is called rather than re-testing the result of that call